### DISCUSSION TITLE
How to prevent Renovate from updating dependencies which are not compatible with other dependencies?

### DISCUSSION THREAD

**Discussion by KevinDW-Fluxys** [renovatebot/renovate#20532] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest (npx)

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have implemented Renovate, but am stumbling on the following case (for nuget packages):

Package A: AutoFixture.AutoNSubstitute 4.17.0 Dependencies: NSubstitute (>= 2.0.3 && < 5.0.0)

Package B: NSubstitute 4.4.0

Renovate tries to upgrade NSubstitute to 5.0.0, but that would break AutoFixture.AutoNSubstitute.

### What you tried so far.

I know i can block the NSubstitute package to 4.4.0 in the renovate.json file with a packagerule, but that would require me to update this rule each time AutoFixture.AutoNSubstitute changes.

Is there a way to configure renovate so that it takes this relation into account?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
not yet, you can workaround with dashboard approval for major updates.

**Reply by KevinDW-Fluxys** [reply]
Unfortunately I'm using Azure Devops which doesn't have dashboard support. I guess we will have to choose between locking manually in config or keeping the PR open as a reminder.

When you say not yet, does it mean something is planned/ongoing for this?

**Reply by viceice** [reply]
maybe some day. currently only platform requirements are in progress. eg nodejs version compabillity for npm packages or python version compabillity for pypi packages.

**Reply by KevinDW-Fluxys** [reply]
ok, thanks for the info, marking as answered

**Reply by RichiCoder1** [reply]
> maybe some day. currently only platform requirements are in progress. eg nodejs version compabillity for npm packages or python version compabillity for pypi packages.

Is there an issue I can follow for this? Running into this today with the package [`sst`](https://github.com/serverless-stack/sst/blob/master/packages/sst/package.json) which requires that workspace versions exactly match it's dependencies for specific packages. Working around this today by ignoring deps and running a provided update command, but would love to ditch it.
